newton/_src/sim/builder.py

```diff
@@ -483,6 +484,106 @@ def __init__(self, up_axis: AxisType = Axis.Z, gravity: float = -9.81):
         self.equality_constraint_polycoef = []
         self.equality_constraint_key = []
         self.equality_constraint_enabled = []
+        # Custom attributes (user-defined per-frequency arrays)
+        # name -> {"assignment": str, "frequency": str, "dtype": wp.dtype, "default": Any, "values": dict[int, Any]}
+        self.custom_attributes: dict[str, dict[str, object]] = {}
+
+    @staticmethod
+    def _default_for_dtype(d: object) -> Any:
+        """Get default value for dtype when not specified."""
+        # vectors default to zeros of their length
+        if wp.types.type_is_vector(d):
+            length = getattr(d, "_shape_", (1,))[0] or 1
+            return np.zeros(
+                length,
+                dtype=wp.types.warp_type_to_np_dtype.get(getattr(d, "_wp_scalar_type_", wp.float32), np.float32),
+            )
+        # scalars
+        if d is wp.bool:
+            return False
+        if d in (wp.int8, wp.int16, wp.int32, wp.int64, wp.uint8, wp.uint16, wp.uint32, wp.uint64):
+            return 0
+        return 0.0
+
+    @staticmethod
+    def _build_wp_array(count: int, d: object, default_val, overrides: dict[int, Any], requires_grad: bool) -> wp.array:
+        """Build wp.array from count, dtype, default and overrides."""
```

Comment:
I don't know if this works but have you tried something simple like this:
```py
arr = [overrides.get(i, default_val) for i in range(count)]
return wp.array(arr, dtype=d, requires_grad=requires_grad)
```
It seems like a lot of work is going on with these type mappings here that seems like a maintenance burden we would rather have Warp take care of.